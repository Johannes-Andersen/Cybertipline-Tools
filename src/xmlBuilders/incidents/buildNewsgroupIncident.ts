import { XMLBuilder, XMLParser } from 'fast-xml-parser';
import type { NewsgroupIncident } from '../../types';
import { buildEmail } from '../common/buildEmail';

const parser = new XMLParser({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  allowBooleanAttributes: true,
});

const builder = new XMLBuilder({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  suppressBooleanAttributes: false,
});

export const buildNewsgroupIncident = (
  newsgroupIncident: NewsgroupIncident,
  keyName = 'newsgroupIncident',
): string =>
  builder.build({
    [keyName]: {
      name: newsgroupIncident.name,
      emailAddress: newsgroupIncident.emailAddress?.map(
        (e) => parser.parse(buildEmail(e)).email,
      ),
      content: newsgroupIncident.content,
      additionalInfo: newsgroupIncident.additionalInfo,
    },
  });
