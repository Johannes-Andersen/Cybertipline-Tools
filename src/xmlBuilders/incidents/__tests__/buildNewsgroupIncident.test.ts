import { describe, expect, it } from 'vitest';
import type { NewsgroupIncident } from '../../../types';
import { buildNewsgroupIncident } from '../buildNewsgroupIncident';

describe('buildNewsgroupIncident', () => {
  it('should build a newsgroup incident', () => {
    const newsgroupIncident: NewsgroupIncident = {
      name: 'name',
      emailAddress: [
        {
          email: 'test@test.com',
          verified: true,
        },
        {
          email: 'test2@test.com',
          verified: false,
        },
      ],
      content: 'content',
      additionalInfo: 'additional info',
    };

    const result = buildNewsgroupIncident(newsgroupIncident);

    expect(result).toMatchSnapshot();
  });
});
