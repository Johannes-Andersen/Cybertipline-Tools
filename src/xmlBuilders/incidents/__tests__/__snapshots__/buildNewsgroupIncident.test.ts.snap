// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildNewsgroupIncident > should build a newsgroup incident 1`] = `"<newsgroupIncident><name>name</name><emailAddress verified="true">test@test.com</emailAddress><emailAddress verified="false">test2@test.com</emailAddress><content>content</content><additionalInfo>additional info</additionalInfo></newsgroupIncident>"`;
